rex_xai/evaluation.py

```diff
@@ -19,16 +19,18 @@ def __init__(self, explanation: Explanation) -> None:
     def ratio(self) -> float:
         """Returns percentage of data required for sufficient explanation"""
         assert self.explanation.explanation is not None
+        assert self.explanation.data.model_channels is not None
```

Comment:
Yeah, I realise this is not ideal. In practice this situation should never occur as it would cause issues much earlier (unless an interactive user is doing something very weird). I added the assert statement to stop the type checker complaining, following the existing code (or I could have added `type: ignore` statements). But I've now changed this to raise an exception instead of the two assert statements.

There's a few different places where this approach is used to stop the type checker complaining, maybe needs a systematic review!